docs/migrate-with-crucible.md

```diff
@@ -0,0 +1,172 @@
+downstairs regions using the `--import-path` flag to specify where the OS image
+is. (See the crucible documentation for information about other flags.)
+
+For example: 
+```
+$ ./target/release/crucible-downstairs create --import-path /home/jordan/images/helios-generic-ttya-base_20230109.raw --data region8810 --uuid $(uuidgen) --extent-size 64000 --extent-count 64
+
+$ ./target/release/crucible-downstairs create --import-path /home/jordan/images/helios-generic-ttya-base_20230109.raw --data region8820 --uuid $(uuidgen) --extent-size 64000 --extent-count 64
+
+$ ./target/release/crucible-downstairs create --import-path /home/jordan/images/helios-generic-ttya-base_20230109.raw --data region8830 --uuid $(uuidgen) --extent-size 64000 --extent-count 64
+```
+
+Each `create` will setup a region file. In the above example, these files are
+`region8810`, `region8820`, and `region8830`, respectively.
+
+### Run the crucible downstairs
+
+After seeding the downstairs with the image, run the downstairs processes.
+
+For example:
+```
+$ ./target/release/crucible-downstairs run -d region8810 -p 8810 -a 172.20.3.73
+$ ./target/release/crucible-downstairs run -d region8820 -p 8820 -a 172.20.3.73
+$ ./target/release/crucible-downstairs run -d region8830 -p 8830 -a 172.20.3.73
+```
+
+### Create a JSON file with disk requests
+
+Now that we've got a crucible volume setup, we need to configure propolis to be
+aware of it as a backend. One can do this by passing the `--crucible-disks`
+flag and a JSON file of an array of `DiskRequest`s  when creating or migrating
+a VM.
+
+On the source machine, create a JSON file like this:
+
+```
+[
+{ 
+    "device": "virtio", 
+    "name": "helios-blockdev", 
+    "read_only": false, 
+    "slot": 1,
+    "volume_construction_request": { 
+        "type": "volume",
+        "block_size": 512, 
+        "id": "0cedae45-3d6e-4d90-b2cb-56f1a1a42a89", 
+        "read_only_parent": null, 
+        "sub_volumes": [
+            { 
+                "type": "region",
+                "block_size": 512, 
+                "blocks_per_extent": 64000, 
+                "extent_count": 64, 
+                "gen": 1, 
+                "opts": { 
+                    "cert_pem": null, 
+                    "control": null, 
+                    "flush_timeout": null, 
+                    "id": "0cedae45-3d6e-4d90-b2cb-56f1a1a42a89", 
+                    "key": null,
+                    "key_pem": null, 
+                    "lossy": false, 
+                    "read_only": false, 
+                    "root_cert_pem": null, 
+                    "target": ["172.20.3.73:8810", 
+                             "172.20.3.73:8820", 
+                             "172.20.3.73:8830" 
+                    ]
+                } 
+            }
+        ]
+    }   
+}   
+]
+```
+
+The `target` field should match where the crucible downstairs processes are
+running. The `blocks_per_extent` field should match what was used when creating
```

Comment:
The (2x) `block_size`, `blocks_per_extent`, and `extent_count` all need to match what was created above.  The `target` is an array of IP:Port where the downstairs are expected to be running.

Reply:
done